### PR TITLE
chore: upgrade oclif to v2

### PR DESCRIPTION
**Pull request by thomaswhyyou**
### Description
Oclif, the cli framework library we use, had a major [version bump to v2](https://oclif.io/blog/2022/01/12/announcing-oclif-v2) earlier this year. 

It included a few breaking changes, which are explained in the migration guide here: https://github.com/oclif/core/blob/main/MIGRATION.md

Some notable changes captured in this PR:
* Args are now configured using `Args` instead of a list of objects (i.e. similar to how flags are configured). 
* `CliUx` module is renamed to `ux`. 
* Some type definitions were removed from the export, so we now add some additional type definitions for `Props` in `base-command.ts`.
* "Global flags" is renamed to "Base flags".

This PR contains no code changes outside of the oclif v2 upgrade. 
 
### Tasks
[KNO-3195](https://linear.app/knock/issue/KNO-3195/[cli]-upgrade-all-deps-to-its-latest-major-versions)

### CODE REVIEW

File: src/commands/workflow/run.ts [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
       summary: "The environment in which to run the workflow",
     }),
-    recipients: commaSeparatedStr({
+    recipients: Flags.string({
       required: true,
       aliases: ["recipient"],
       summary:
         "One or more recipient ids for this workflow run, separated by comma.",
+      multiple: true,
+      delimiter: ",",
```
Comment (thomaswhyyou) [inline src/commands/workflow/run.ts:24]: @meryldakin Sooo it turns out oclif v2 added a `delimiter` option that would parse the string value into an array, which is the same thing we were doing with `commaSeparatedStr` 😅 (but you have to set `multiple: true`). 

I removed the helper in favor of using the built-in delimiter option, but wanted to give you a heads up!

File: src/lib/base-command.ts [hunk 2]
```diff
@@ -27,8 +53,8 @@ abstract class BaseCommand extends Command {
     this.runContext = await RunContext.load(this.id);
   }
 
-  // Global flags are inherited by any command that extends BaseCommand.
-  static globalFlags = {
+  // Base flags are inherited by any command that extends BaseCommand.
+  static baseFlags = {
```
Comment (thomaswhyyou) [inline src/lib/base-command.ts:57]: `globalFlags` has been renamed to `baseFlags`.

File: test/commands/translation/push.test.ts [hunk 3]
```diff
@@ -232,9 +232,7 @@ describe("commands/translation/push", () => {
           const stub2 = KnockApiV1.prototype.upsertTranslation as any;
           sinon.assert.calledThrice(stub2);
 
-          const expectedArgs = {
-            translationRef: undefined,
-          };
+          const expectedArgs = {};
```
Comment (thomaswhyyou) [inline test/commands/translation/push.test.ts:235]: It appears the behavior changed with args when an arg was defined but not provided — before we used to get `{arg: undefined}` vs nothing at all.

File: src/commands/workflow/get.ts [hunk 4]
```diff
@@ -96,7 +100,7 @@ export default class WorkflowGet extends BaseCommand {
     this.log("");
 
     if (workflow.steps.length === 0) {
-      return CliUx.ux.log(" This workflow has no steps to display.");
+      return ux.log(" This workflow has no steps to display.");
```
[the remark this answers is not in the document]
Comment (thomaswhyyou) [reply src/commands/workflow/get.ts:103]: Ah the leading space is there intentionally to align the left spacing in the display. Will add a quick comment.

File: src/commands/workflow/run.ts (hunk 1 above)
Comment (meryldakin) [reply src/commands/workflow/run.ts:24]: totally missed this!! gotcha thanks!

**Pull request merged**